skrf/vi/vna/hp/hp8510c_sweep_plan.py

```diff
@@ -0,0 +1,248 @@
+import dataclasses
+import numpy as np
+from typing import List, Union
+from abc import ABC, abstractmethod
+
+
+class SweepSection(ABC):
+    @abstractmethod
+    def get_hz(self):
+        ''' List of hz represented by this section after applying mask '''
+        pass
+
+    def get_raw_hz(self):
+        ''' List of hz fetched from the instrument before applying mask '''
+        return self.get_hz()
+
+    @abstractmethod
+    def apply_8510(self, hp8510c):
+        pass
+
+    def mask_8510(self, network):
+        return network
+
+
+@dataclasses.dataclass
+class LinearBuiltinSweepSection(SweepSection):
+    hz_min: float
+    hz_max: float
+    n_points: int
+
+    def get_hz(self):
+        return np.linspace(self.hz_min, self.hz_max, self.n_points)
+
+    def apply_8510(self, hp8510c):
+        hp8510c._set_instrument_step_state(self.hz_min, self.hz_max, self.n_points)
+
+
+@dataclasses.dataclass
+class LinearMaskedSweepSection(SweepSection):
+    hz_min: float
+    hz_max: float
+    n_points: int
+    mask: object
```

Comment:
Is there a more specific type than `object`?

Reply:
Whoops I forgot that one on my first commit, second commit incoming.